base/loading.jl

```diff
@@ -490,155 +435,128 @@ end
+                if deps isa Vector
+                    found_name = name in deps
+                    break
+                elseif deps isa Dict
+                    for (dep, uuid) in deps
+                        if dep === name
+                            return PkgId(UUID(uuid), name)
+                        end
+                    end
                 end
             end
         end
-        # now search through `deps = []` string to see if we have an entry for `name`
-        uuid == where || return false
-        deps === nothing && return true
-        # TODO: handle inline table syntax
-        if deps[1] != '[' || deps[end] != ']'
-            @warn "Unexpected TOML deps format:\n$deps"
-            return false
-        end
-        occursin(repr(name), deps) || return true
-        seekstart(io) # rewind IO handle
-        # finally, find out the `uuid` associated with `name`
-        return something(manifest_file_name_uuid(io, name), false)
     end
-    found_or_uuid isa UUID && return PkgId(found_or_uuid, name)
-    found_or_uuid && return PkgId(name)
-    return nothing
+    found_where || return nothing
+    found_name || return PkgId(name)
+    # Only here is deps was not a dict which mean we have a unique name for the dep
+    if !haskey(d, name) || length(d[name]) != 1
+        error("expected a single entry for $(repr(name)) in $(repr(project_file))")
+    end
+    entry = first(d[name])
+    if found_name
+        uuid = get(entry, "uuid", nothing)
+        return PkgId(UUID(uuid), name)
+    else
+        return PkgId(name)
+    end
 end
 
 # find `uuid` stanza, return the corresponding path
-function explicit_manifest_uuid_path(project_file::String, pkg::PkgId)::Union{Nothing,String}
-    manifest_file = project_file_manifest_path(project_file)
+function explicit_manifest_uuid_path(project_file::String, pkg::PkgId, cache::TOMLCache)::Union{Nothing,String}
+    manifest_file = project_file_manifest_path(project_file, cache)
     manifest_file === nothing && return nothing # no manifest, skip env
-    open(manifest_file) do io
-        uuid = name = path = hash = nothing
-        for line in eachline(io)
-            if (m = match(re_section_capture, line)) !== nothing
-                uuid == pkg.uuid && break
-                name = String(m.captures[1])
-                path = hash = nothing
-            elseif (m = match(re_uuid_to_string, line)) !== nothing
-                uuid = UUID(m.captures[1])
-            elseif (m = match(re_path_to_string, line)) !== nothing
-                path = String(m.captures[1])
-            elseif (m = match(re_hash_to_string, line)) !== nothing
-                hash = SHA1(m.captures[1])
+
+    d = parsed_toml(cache, manifest_file)
+    entries = get(d, pkg.name, nothing)
+    entries === nothing && return nothing # TODO: allow name to mismatch?
+    for entry in entries
+        if UUID(get(entry, "uuid", nothing)) === pkg.uuid
```

Comment:
`ERROR: MethodError: Cannot `convert` an object of type Nothing to an object of type UInt128`